### DISCUSSION TITLE
Can't set custom label on object field using ui : itemProps(item)

### DISCUSSION THREAD

**Discussion by NicolasRoehm** [tinacms/tinacms#4134] [Ideas]
### Describe the issue that you're seeing. Any Loom videos or screenshots usually help a lot!

I'm trying to set a custom label on an `object` field using `ui : itemProps(item)` but the label remains `main_field`.
Looks like it's only working for `list:true`
```ts
export default defineConfig({
  schema : {
    collections: [
      {
        name : 'test_collection',
        label : 'Test collection',
        path  : 'content/tests',
        fields : [
          {
            name  : 'main_field',
            type  : 'object',
            ui    : {
              itemProps(item) {
                return { label : item?.sub_field || 'Custom Label' };
              },
            },
            fields : [ { name : 'sub_field', type : 'string' }]
          }
        ]
      }
    ]
  }
});
```

### Reproduction

https://github.com/NicolasRoehm/tina-app/blob/53438839aa77d579fd646c613c630e35917c113a/tina/config.js#L13

### Steps to reproduce

- Fresh install using `npx create-tina-app@latest` but also visible at latest
  - tinacms : 1.5.17
  - @tinacms/cli : 1.5.26
- Described config [tina/config.js](https://github.com/NicolasRoehm/tina-app/blob/main/tina/config.js#L13)

### System Info

```shell
npx envinfo --system --npmPackages '{tinacms,@tinacms/*}' --binaries --browsers
Need to install the following packages:
  envinfo@7.10.0
Ok to proceed? (y) y

  System:
    OS: Windows 10 10.0.19045
    CPU: (12) x64 Intel(R) Core(TM) i7-10850H CPU @ 2.70GHz
    Memory: 15.59 GB / 31.61 GB
  Binaries:
    Node: 18.14.2 - C:\laragon\bin\nodejs\node-v18.14.2-win-x64\node.EXE
    Yarn: 1.22.19 - C:\code\next-boilerplate\node_modules\.bin\yarn.CMD 
    npm: 9.3.0 - ~\AppData\Roaming\npm\npm.CMD
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (113.0.1774.35)
    Internet Explorer: 11.0.19041.1566
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/tinacms/tinacms/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [docs](https://tina.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/tinacms/tinacms/issues) that reports the same bug.

**Comment by jamespohalloran** [comment]
Yeah itemProps is by design intended to only work with list fields, as a way of distinguishing between the different list items (so this is more of a feature request than a bug). I'm going to convert this to a discussion. 

For extra context; in your use-case, why do you need that group label to be configurable?

**Reply by smiley-uriux** [reply]
I landed here looking for this as well. My use case is an "Address" field (street, city state, etc) that I am using an object field for. It's way more useful to display some shortened version of the address instead of something generic like "Address" which requires me to click into the field to see what it is currently set to.

**Reply by NicolasRoehm** [reply]
Exactly, personally I'm look for something like this 
![tina-object-label](https://github.com/tinacms/tinacms/assets/6847846/d610deeb-1b8d-42d5-9d47-85e9e9a64438)

**Reply by jayalfredprufrock** [reply]
Ideally this could support mdx components/template as well. Currently when embedding MDX components into markdown body, the RTE just shows the name of the component. This can make it hard to distinguish between multiple uses of the same component within the markdown, especially since I'm guessing the "click to edit" functionality is not likely to ever support MDX embeds. Ideally we could leverage the itemProps callback to dynamically render the label that is displayed within the RTE so we can add extra contextual info to the label.

**Reply by freshyill** [reply]
I was nearly pulling my hair out trying to figure this out.

In addition to preview of the object's content, there really needs to be a label on the _object_, to be consistent with the rest of the form. If I'm a user scanning a for a particular thing to edit within a long form (whether it's a field or object), I'm going to be looking for _labels_ first, not the value of the field (or object).

To a user, everything in the form is a field. Some just happen to open up to reveal more fields. It's jarring that some fields don't have proper labels. As someone _configuring_ the CMS, it's jarring _to me_ that the label I specified doesn't appear where all other labels appear.
